subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -119,6 +119,65 @@ static inline void set_pkt_ll_addr(struct net_linkaddr *addr, bool comp,
 	addr->type = NET_LINK_IEEE802154;
 }
 
+/**
+ * Filters the destination address of the frame (used when IEEE802154_HW_FILTER
+ * is not available).
+ */
+static bool ieeee802154_check_dst_addr(struct net_if *iface,
+				      struct ieee802154_mhr *mhr)
+{
+	struct ieee802154_context *ctx = net_if_l2_data(iface);
+	uint8_t i;
+
+	/*
+	 * Apply filtering requirements from chapter 6.7.2 of the IEEE
+	 * 802.15.4-2015 standard:
+	 */
+
+	if (mhr->fs->fc.dst_addr_mode == IEEE802154_ADDR_MODE_NONE) {
+		/* TODO: apply d.4 and d.5 when PAN coordinator gets implemented */
+		/* also, macImplicitBroadcast is not implemented */
+		return false;
+	}
+
+	/*
+	 * c. If a destination PAN ID is included in the frame, it shall match
+	 * macPanId or shall be the broadcastPAN ID
+	 */
+	if (!(mhr->dst_addr->plain.pan_id == 0xFFFF
```

Comment:
nit: I think it would be better to define something like `BROADCAST_ADDRESS` instead of using magic numbers.

Reply:
Yep, got it, they are defined inside ieee802154_frame.h ;)